### PR TITLE
Add gp_read_error_log and gp_truncate_error_log functions to the cata…

### PR DESCRIPTION
**Pull request by asubramanya**
…log.

gp_read_error_log and gp_truncate_error_log were two functions which were added
in 4.3.3.0 to support the new error logging functionality for external tables.
They were added to a "hidden" catalog in order to prevent bumping up the
catalog version in a minor version release. We need to add them to the true
catalog in a major release.

Also in the process remove hiddencat mechanism since apart from the above two
functions no other functions use it.

### CODE REVIEW

**Comment by ashwinstar** [comment]
Looks good to me.

File: src/include/catalog/pg_proc.h [hunk 1]
```diff
@@ -4133,13 +4133,15 @@ DESCR("I/O");
...
 
 /*
- *  These are dummy at this point until the next catalog update, but need to be here
- *  for fmgr generator script as well as unused_oids.
-
-DATA(insert OID = 3000 (gp_read_error_log PGNSP PGUID 12 0 f f t t v 1 2249 f "25" _null_ _null_ _null_ gp_read_error_log - _null_ r ));
-DATA(insert OID = 3069 (gp_truncate_error_log PGNSP PGUID 12 0 f f t t v 1 16 f "25" _null_ _null_ _null_ gp_truncate_error_log - _null_ r ));
+ * Ideally this function definition should be in pg_proc.sql and should be
+ * generated by running catullus.pl script. But since we cannot specify the
+ * PRODATAACCESS column of pg_proc as SEGMENT ('s') using the CREATE FUNCTION
```
Comment (asimrp) [inline src/include/catalog/pg_proc.h]: A bigger issue is that PRODATAACCESS value of `s` is not accessible to users.  Yet, they will be able to see it when selecting from `pg_proc`.  I think we should augment parser so that users can use `s` in `CREATE FUNCTION`.  The machinery to implement `s` is already present.  But this can go in as a separate commit.

File: src/include/catalog/pg_proc.h [hunk 2]
```diff
@@ -4133,13 +4133,15 @@ DESCR("I/O");
...
+ * syntax, we need to create the DATA statement manually and place it here.
+ */
 
-*/
+/* gp_read_error_log(IN exttable text, OUT cmdtime timestamptz, OUT relname text, OUT filename text, OUT linenum int4, OUT bytenum int4, OUT errmsg text, OUT rawdata text, OUT rawbytes bytea) => SETOF record */ 
+DATA(insert OID = 3000 ( gp_read_error_log  PGNSP PGUID 12 0 f f t t v 1 2249 f "25" "{25,1184,25,25,23,23,25,25,17}" "{i,o,o,o,o,o,o,o,o}" "{exttable,cmdtime,relname,filename,linenum,bytenum,errmsg,rawdata,rawbytes}" gp_read_error_log _null_ _null_ s ));
```
Comment (asimrp) [inline src/include/catalog/pg_proc.h]: As Ashwin has already pointed out, this being Greenplum specific, belongs in `pg_proc_gp.h`.

**Comment by asimrp** [comment]
Thank you for the clean up!  :+1:

File: src/include/catalog/pg_proc.h (hunk 2 above)
Comment (asubramanya) [inline src/include/catalog/pg_proc.h]: Manually adding the `DATA` statement in `pg_proc_gp.h` will not work because `pg_proc_gp.h` is generated by `catullus.pl`. The script uses function definitions in `pg_proc.sql` in order to generate the header file and will rewrite the entire header file based on its contents.
